File: app/src/main/java/com/developer/valyutaapp/adapter/WidgetAdapter.java
package com.developer.valyutaapp.adapter;

import android.content.Context;
import android.graphics.Bitmap;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.CheckBox;
import android.widget.CompoundButton;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.ListView;
import android.widget.TextView;


import com.developer.valyutaapp.R;
import com.developer.valyutaapp.model.Valute;
import com.developer.valyutaapp.utils.ImageResource;

import java.util.ArrayList;

public class WidgetAdapter extends BaseAdapter {

    private ClickListener clickListener;
    Context context;
    ArrayList<Valute> valutes;

    public WidgetAdapter(Context context, ArrayList<Valute> valutes) {
        this.context = context;
        this.valutes = valutes;
    }

    @Override
    public int getCount() {
        return valutes.size();
    }

    @Override
    public Object getItem(int i) {
        return valutes.get(i);
    }

    @Override
    public long getItemId(int i) {
        return valutes.get(i).getId();
    }

    @Override
    public View getView(final int i, View view, ViewGroup viewGroup) {
        LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        View v = inflater.inflate(R.layout.sort_item, viewGroup, false);
        LinearLayout linearLayout = (LinearLayout) v.findViewById(R.id.item_dialog);
        ListView listView = (ListView) v.findViewById(R.id.list_dialog);
        ImageView icon = (ImageView) v.findViewById(R.id.img_flag);
        final CheckBox checkBox = (CheckBox) v.findViewById(R.id.checkSort);
        TextView tvName = (TextView) v.findViewById(R.id.name_currency);

        Bitmap bt = ImageResource.getImageRes(context, String.valueOf(valutes.get(i).getCharCode()));
        icon.setImageBitmap(bt);
        tvName.setText(String.valueOf(valutes.get(i).getName()));

        checkBox.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(CompoundButton compoundButton, boolean b) {
                if (checkBox.isChecked()) {
                    checkBox.setChecked(true);
                } else {
                    checkBox.setChecked(false);
                }
            }
        });

        linearLayout.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                clickListener.itemClicked(valutes.get(i), i);
            }
        });
        return v;
    }

    public void setClickListener(ClickListener clickListener) {
        this.clickListener = clickListener;
    }

    public interface ClickListener {
        void itemClicked(Valute item, int position);
    }
}

File: app/src/main/java/com/developer/valyutaapp/ui/adapter/WidgetAdapter.kt
package com.developer.valyutaapp.ui.adapter

import android.annotation.SuppressLint
import android.content.Context
import com.developer.valyutaapp.domain.entities.Valute
import android.view.ViewGroup
import android.view.LayoutInflater
import com.developer.valyutaapp.R
import android.view.View
import android.widget.*
import com.developer.valyutaapp.utils.ImageResource

class WidgetAdapter(
    private val context: Context,
    private val valutes: MutableList<Valute>,
    private val onItemValuteClick: (Valute, Int) -> Unit,
) : BaseAdapter() {
    override fun getCount(): Int {
        return valutes.size
    }

    override fun getItem(i: Int): Any {
        return valutes[i]
    }

    override fun getItemId(i: Int): Long {
        return valutes[i].id.toLong()
    }

    @SuppressLint("ViewHolder")
    override fun getView(i: Int, view: View, viewGroup: ViewGroup): View {
        val inflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        val v = inflater.inflate(R.layout.favorites_item, viewGroup, false)
        val linearLayout = v.findViewById<View>(R.id.item_dialog) as LinearLayout
        //val listView = v.findViewById<View>(R.id.list_dialog) as ListView
        val icon = v.findViewById<View>(R.id.img_flag) as ImageView
        val checkBox = v.findViewById<View>(R.id.favorite) as ImageView
        val tvName = v.findViewById<View>(R.id.name_currency) as TextView
        val bt = ImageResource.getImageRes(context, valutes[i].charCode)
        icon.setImageBitmap(bt)
        tvName.text = valutes[i].name
//        checkBox.setOnCheckedChangeListener { _, _ ->
//            checkBox.isChecked = checkBox.isChecked
//        }
        linearLayout.setOnClickListener {
            onItemValuteClick(valutes[i], i)
        }
        return v
    }
}